pkg/cli/admin/nodeimage/create.go

```diff
@@ -116,17 +138,36 @@ type CreateOptions struct {
 	AssetsDir string
 	// OutputName allows the user to specify the name of the generated image.
 	OutputName string
+	// Simpler interface for creating a single node
+	SingleNodeOpts *singleNodeCreateOptions
 
 	nodeJoinerExitCode int
 	rsyncRshCmd        string
 }
 
+type singleNodeCreateOptions struct {
+	MacAddress      string
+	CPUArchitecture string
+	SSHKey          string
+	Hostname        string
+	RootDeviceHints string
+	NetworkConfig   string
+}
+
 // AddFlags defined the required command flags.
 func (o *CreateOptions) AddFlags(cmd *cobra.Command) {
 	flags := o.addBaseFlags(cmd)
 
 	flags.StringVar(&o.AssetsDir, "dir", o.AssetsDir, "The path containing the configuration file, used also to store the generated artifacts.")
 	flags.StringVarP(&o.OutputName, "output-name", "o", "node.iso", "The name of the output image.")
+
+	flags.StringP(snFlagMacAddress, "m", "", "Single node flag. MAC address used to identify the host to apply the configuration. If specified, the nodes-config.yaml config file will not be used.")
```

Comment:
I think we should omit this flag and apply any config given on the command line unconditionally to whichever host gets booted with the ISO. (I recognise that this requires changes to the installer and likely our client for assisted-service - perhaps we can take the opportunity to make the same improvement for SNO on day 1.) This would relieve the user of the need to find out the MAC address of the host, which can be a giant pain, without any loss of functionality.

Reply:
Removing the mac address flag will require changes to the installer. There is [a validation](https://github.com/openshift/installer/blob/fe126cafe857fa5e4e7f88a2134a19ae2f089b85/pkg/asset/agent/agentconfig/agenthosts.go#L156) that checks that at least one MAC address is specified for each host. I like the idea of making the MAC address optional, but it would require more planning.